components/brave_today/browser/brave_news_p3a.cc

```diff
@@ -65,6 +125,27 @@ void RecordWeeklySessionCount(PrefService* prefs, bool is_add) {
                           total_session_count);
 }
 
+void ResetCurrSessionTotalViewsCount(PrefService* prefs) {
+  prefs->SetUint64(prefs::kBraveTodayCurrSessionCardViews, 0);
+  VLOG(1) << "NewsP3A: reset curr session total card views count";
+}
+
+}  // namespace
+
+void RecordAtSessionStart(PrefService* prefs) {
+  base::Time now_midnight = base::Time::Now().LocalMidnight();
+  prefs->SetTime(prefs::kBraveTodayLastSessionTime, now_midnight);
+  if (prefs->GetTime(prefs::kBraveTodayFirstSessionTime).is_null()) {
+    prefs->SetTime(prefs::kBraveTodayFirstSessionTime, now_midnight);
+  }
```

Comment:
FWIW, you can also `prefs->IsDefaultValue()` to check if a pref has been previously set. I don't think there's any particular advantage one way or the other here, though. Whichever you prefer.

Reply:
good to know, will keep this in my arsenal for the future, thx for the tip